<a href="https://colab.research.google.com/github/leeryleprechaun/MSDS_Final_Project/blob/main/Week_3_Financial_Data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [79]:
# Loading in packages and the raw date
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline



In [80]:
# Loading in THE ORIGINAL C6 data, I want to take a look at the "Relevant, Calculated" Columns againt
url='https://drive.google.com/file/d/1IUydVVPtD1AMlUoWzCHDa6PzxbYFd2BD/view?usp=sharing'
url='https://drive.google.com/uc?id=' + url.split('/')[-2]
df_C6 = pd.read_csv(url)
df_C6.head(2)

Account number                                       Organization Country  \
0          843435  (ACIP) Alexandria Company for Industrial Packages   Egypt   
1          843435  (ACIP) Alexandria Company for Industrial Packages   Egypt   

   Public Response received date  Primary activity  \
0  public                57:20.0  Plastic products   
1  public                57:20.0  Plastic products   

                  Primary sector Primary industry  \
0  Plastic product manufacturing    Manufacturing   
1  Plastic product manufacturing    Manufacturing   

  Primary questionnaire sector  Row                       RowName  \
0                      General    1  Purchased goods and services   
1                      General    2                 Capital goods   

                              C6_5_C1                  C6_5_C2  \
0        Relevant, not yet calculated  Question not applicable   
1  Not relevant, explanation provided  Question not applicable   

                   C6_5_C3                  C6_5_C4  \
0  Question not applicable  Question not applicable   
1  Question not applicable  Question not applicable   

                                             C6_5_C5  
0  Due to the recent establishment of ACIP's sust...  
1  The percentage of capital goods Scope 3 emissi...

In [81]:
# Creating a list that has all the companies just listed one time

# First I will trim out the data columns that I do not need
df_trim = df_C6
df_trim.drop(['Public','Response received date', 'Response received date', 'Primary questionnaire sector'],axis=1,inplace=True)



In [82]:
# How many companies are in the list???
df_trim.nunique()

Account number       5913
Organization         5913
Country               102
Primary activity      197
Primary sector         58
Primary industry       13
Row                    17
RowName                17
C6_5_C1                 6
C6_5_C2             14458
C6_5_C3             16314
C6_5_C4               506
C6_5_C5             33522
dtype: int64

In [83]:
# Pivot the date frame
pivot_df = df_trim.pivot(index=('Account number', 'Organization', 'Country', 'Primary activity', 'Primary sector', 'Primary industry'), columns='RowName', values = ['C6_5_C2'])
pivot_df

C6_5_C2  \
RowName                                                                                                                                                                                                                                           Business travel   
Account number Organization                                       Country                                            Primary activity                   Primary sector                                     Primary industry                                         
44             3i Group                                           United Kingdom of Great Britain and Northern Ir... Asset managers                     Financial services                                 Services                                          40.2   
53             Abertis Infraestructuras                           Spain                                              Infrastructure upkeep & management Land & property ownership & development            Infrastructure                                22581.41   
58             ABM INDUSTRIES INC                                 United States of America                           Commercial services                Commercial & consumer services                     Services                                       2195.01   
64             Abbott Laboratories                                United States of America                           Medical equipment                  Medical equipment & supplies                       Biotech, health care & pharma                    68909   
79             A T Kearney Ltd                                    United Kingdom of Great Britain and Northern Ir... Other professional services        Specialized professional services                  Services                                      21714.11   
...                                                                                                                                                                                                                                                           ...   
863233         CITIC TOURISM GROUP CO.,LTD.                       China                                              Travel services                    Commercial & consumer services                     Services                                           200   
863244         VAMOS LOCACAO DE CAMINHOES, MAQUINAS E EQUIPAME... Brazil                                             Transportation support services    Industrial support services                        Services                       Question not applicable   
863247         Aeris Energy SA                                    Brazil                                             Other renewable energy equipment   Renewable energy equipment                         Manufacturing                  Question not applicable   
863343         PE Agrarkontor                                     Germany                                            Animal products wholesale          Trading, wholesale, distribution, rental & leasing Retail                         Question not applicable   
863355         Euro Packaging UK Ltd                              United Kingdom of Great Britain and Northern Ir... Accessories                        Leisure & home manufacturing                       Manufacturing                  Question not applicable   

                                                                                                                                                                                                                                                                   \
RowName                                                                                                                                                                                                                                             Capital goods   
Account number Organization                                       Country

In [84]:
# A count of all the values
pivot_df.count()

         RowName                                                          
C6_5_C2  Business travel                                                      5884
         Capital goods                                                        5886
         Downstream leased assets                                             5894
         Downstream transportation and distribution                           5884
         Employee commuting                                                   5883
         End of life treatment of sold products                               5895
         Franchises                                                           5892
         Fuel-and-energy-related activities (not included in Scope 1 or 2)    5886
         Investments                                                          5891
         Other (downstream)                                                   5902
         Other (upstream)                                                     5899
         Pro

In [85]:
pivot_df.replace('Question not applicable', 'NaN')



pivot_df.nunique()

         RowName                                                          
C6_5_C2  Business travel                                                      2244
         Capital goods                                                        1192
         Downstream leased assets                                              355
         Downstream transportation and distribution                           1065
         Employee commuting                                                   1801
         End of life treatment of sold products                                761
         Franchises                                                            122
         Fuel-and-energy-related activities (not included in Scope 1 or 2)    1935
         Investments                                                           307
         Other (downstream)                                                     61
         Other (upstream)                                                       84
         Pro

In [86]:
namesList = ['business_travel', 'capital_goods', 'downstream_assets', 'downstream_trans', 'employee_comute', 'end_of_prod_life', 'franchises','non_scope_1_2_energy','investment','downstream','upstream','precessing_products','purchased_goods_and_services','upstream_assets','upstream_trans','sold_product_use','operations_waste']
pivot_df.columns = pivot_df.columns[:0].tolist() + namesList
pivot_df.head(10)

business_travel  \
Account number Organization                  Country                                            Primary activity                   Primary sector                          Primary industry                                         
44             3i Group                      United Kingdom of Great Britain and Northern Ir... Asset managers                     Financial services                      Services                                          40.2   
53             Abertis Infraestructuras      Spain                                              Infrastructure upkeep & management Land & property ownership & development Infrastructure                                22581.41   
58             ABM INDUSTRIES INC            United States of America                           Commercial services                Commercial & consumer services          Services                                       2195.01   
64             Abbott Laboratories           United States of America                           Medical equipment                  Medical equipment & supplies            Biotech, health care & pharma                    68909   
79             A T Kearney Ltd               United Kingdom of Great Britain and Northern Ir... Other professional services        Specialized professional services       Services                                      21714.11   
97             Aareal Bank AG                Germany                                            Banks                              Financial services                      Services                                           518   
118            Acron                         Russian Federation                                 Nitrogenous fertilizers            Chemicals                               Materials                      Question not applicable   
119            Resolute Forest Products Inc. Canada                                             Pulp & paper mills                 Wood & paper materials                  Materials                                          367   
123            ACI WORLDWIDE  LTD            United Kingdom of Great Britain and Northern Ir... Software                           IT & software development               Services                       Question not applicable   
134            Aboitiz Equity Ventures       Philippines                                        Electricity networks               Energy utility networks                 Infrastructure                 Question not applicable   

                                                                                                                                                                                                                    capital_goods  \
Account number Organization                  Country                                            Primary activity                   Primary sector                          Primary industry                                         
44             3i Group                      United Kingdom of Great Britain and Northern Ir... Asset managers                     Financial services                      Services                       Question not applicable   
53             Abertis Infraestructuras      Spain                                              Infrastructure upkeep & management Land & property ownership & development Infrastructure                 Question not applicable   
58             ABM INDUSTRIES INC            United States of America                           Commercial services                Commercial & consumer services          Services                       Question not applicable   
64             Abbott Laboratories           United States of America                           Medical equipment                  Medical equipment & supplies            Biotech, health care & pharma                  1878527   
79             A T Kearney Ltd               United Kingdom of Great Britain and Northe

In [87]:
# Adding in companies that were dropped from the data frame
# Loading in data
url='https://drive.google.com/file/d/1EoOfPxcBTexmvY42a7L9KPfjDSBeKx7o/view?usp=sharing'
url='https://drive.google.com/uc?id=' + url.split('/')[-2]
code_df = pd.read_csv(url)
code_df

Organization_ID                                               Name  \
0                   1                                         3M Company   
1                   2                                Abbott Laboratories   
2                   2                                Abbott Laboratories   
3                   3                                             AbbVie   
4                   4                                      Accenture plc   
...               ...                                                ...   
3298            19373                      Zhongsheng Group Holdings Ltd   
3299            19374                  Zhuzhou CSR Times Electric Co Ltd   
3300              503                                             Zoetis   
3301            19381  Zoomlion Heavy Industry Science and Technology...   
3302            19384                                ZUG ESTATES HOLDING   

                      source_org_id  
0                             3M Co  
1               Abbott Laboratories  
2     Abbott Laboratories (India) L  
3                        AbbVie Inc  
4                     Accenture plc  
...                             ...  
3298   Zhongsheng Group Holdings Lt  
3299    Zhuzhou CRRC Times Electric  
3300                     Zoetis Inc  
3301   Zoomlion Heavy Industry Scie  
3302         Zug Estates Holding AG  

[3303 rows x 3 columns]

In [88]:
#Save file to google drive

from google.colab import drive

drive.mount('/content/drive')
path = '/content/drive/My Drive/output.csv'

with open(path, 'w', encoding = 'utf-8-sig') as f:
  pivot_df.to_csv(f)


Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [89]:
#Reloading the pivot file
url='https://drive.google.com/file/d/1k02L43h4EZYYE5UJB0U4TqhsYeGYgaQT/view?usp=share_link'
url='https://drive.google.com/uc?id=' + url.split('/')[-2]
pivot_df = pd.read_csv(url)
pivot_df

Account number                                       Organization  \
0                 44                                           3i Group   
1                 53                           Abertis Infraestructuras   
2                 58                                 ABM INDUSTRIES INC   
3                 64                                Abbott Laboratories   
4                 79                                    A T Kearney Ltd   
...              ...                                                ...   
5908          863233                       CITIC TOURISM GROUP CO.,LTD.   
5909          863244  VAMOS LOCACAO DE CAMINHOES, MAQUINAS E EQUIPAM...   
5910          863247                                    Aeris Energy SA   
5911          863343                                     PE Agrarkontor   
5912          863355                              Euro Packaging UK Ltd   

                                                Country  \
0     United Kingdom of Great Britain and Northern I...   
1                                                 Spain   
2                              United States of America   
3                              United States of America   
4     United Kingdom of Great Britain and Northern I...   
...                                                 ...   
5908                                              China   
5909                                             Brazil   
5910                                             Brazil   
5911                                            Germany   
5912  United Kingdom of Great Britain and Northern I...   

                        Primary activity  \
0                         Asset managers   
1     Infrastructure upkeep & management   
2                    Commercial services   
3                      Medical equipment   
4            Other professional services   
...                                  ...   
5908                     Travel services   
5909     Transportation support services   
5910    Other renewable energy equipment   
5911           Animal products wholesale   
5912                         Accessories   

                                         Primary sector  \
0                                    Financial services   
1               Land & property ownership & development   
2                        Commercial & consumer services   
3                          Medical equipment & supplies   
4                     Specialized professional services   
...                                                 ...   
5908                     Commercial & consumer services   
5909                        Industrial support services   
5910                         Renewable energy equipment   
5911  Trading, wholesale, distribution, rental & lea...   
5912                       Leisure & home manufacturing   

                   Primary industry          business_travel  \
0                          Services                     40.2   
1                    Infrastructure                 22581.41   
2                          Services                  2195.01   
3     Biotech, health care & pharma                    68909   
4                          Services                 21714.11   
...                             ...                      ...   
5908                       Services                      200   
5909                       Services  Question not applicable   
5910                  Manufacturing  Question not applicable   
5911                         Retail  Question not applicable   
5912                  Manufacturing  Question not applicable   

                capital_goods        downstream_assets  \
0     Question not applicable  Question not applicable   
1     Question not applicable  Question not applicable   
2     Question not applicable  Question not applicable   
3                     1878527  Question not applicable   
4                           0  Question not applicable   
...                       ...            

In [90]:
url='https://drive.google.com/file/d/1pXPiiiBZqQYxT-LzN9cavN8LcZYGK9NJ/view?usp=sharing'
url='https://drive.google.com/uc?id=' + url.split('/')[-2]
df_revenue = pd.read_csv(url)
df_revenue

Company Name  Data Year  Revenue (in $USD)
0                               ABB Ltd       2020       2.613400e+10
1                                 AECOM       2020       1.323998e+10
2            S & U Plc (United Kingdom)       2020       1.153064e+08
3         George Kent (Malaysia) Berhad       2020       8.111401e+07
4                  Sapura Energy Berhad       2020       1.557767e+09
...                                 ...        ...                ...
8954                          Zovio Inc       2020       3.971210e+08
8955          Zurn Water Solutions Corp       2020       1.433100e+09
8956                      Zymeworks Inc       2020       3.895100e+07
8957                          Zynex Inc       2020       8.012200e+07
8958  Sleep Country Canada Holdings Inc       2020       5.740144e+08

[8959 rows x 3 columns]

In [91]:
#Combining what I currently have and matches based on the current name
#Note that I have three frames right now, pivot_df, df_revenue, and code_df
merge_1 = pd.merge(pivot_df, code_df, left_on='Organization', right_on='Name')
merge_1.head(2)

Account number         Organization                   Country  \
0              58   ABM INDUSTRIES INC  United States of America   
1              64  Abbott Laboratories  United States of America   

      Primary activity                  Primary sector  \
0  Commercial services  Commercial & consumer services   
1    Medical equipment    Medical equipment & supplies   

                Primary industry business_travel            capital_goods  \
0                       Services         2195.01  Question not applicable   
1  Biotech, health care & pharma           68909                  1878527   

         downstream_assets         downstream_trans  ...  \
0  Question not applicable  Question not applicable  ...   
1  Question not applicable  Question not applicable  ...   

                  upstream      precessing_products  \
0  Question not applicable  Question not applicable   
1  Question not applicable  Question not applicable   

  purchased_goods_and_services          upstream_assets  \
0      Question not applicable  Question not applicable   
1                      6580577  Question not applicable   

            upstream_trans         sold_product_use         operations_waste  \
0  Question not applicable  Question not applicable  Question not applicable   
1                  1610768                   508491                    15088   

  Organization_ID                 Name         source_org_id  
0           19388   ABM INDUSTRIES INC  ABM Industries, Inc.  
1               2  Abbott Laboratories   Abbott Laboratories  

[2 rows x 26 columns]

In [92]:
#Merging in the next info
df_rev_merge = pd.merge(merge_1, df_revenue, left_on='source_org_id', right_on='Company Name')
df_rev_merge = df_rev_merge.drop(columns = ['Organization_ID', 'source_org_id', 'Name'])
df_rev_merge.drop(df_rev_merge.columns[[0]], axis = 1)

df_rev_merge

Account number                                Organization  \
0                58                          ABM INDUSTRIES INC   
1                64                         Abbott Laboratories   
2                97                              Aareal Bank AG   
3               119               Resolute Forest Products Inc.   
4               154                                ACCIONA S.A.   
..              ...                                         ...   
930           74141                 Knight-Swift Transportation   
931          827295                       Asahi Tec Corporation   
932          827445  Baycom Opto-Electronics Technology Company   
933          828093                        Daido Kogyo Co.,Ltd.   
934          844147                                       Veeco   

                      Country                 Primary activity  \
0    United States of America              Commercial services   
1    United States of America                Medical equipment   
2                     Germany                            Banks   
3                      Canada               Pulp & paper mills   
4                       Spain    Renewable energy construction   
..                        ...                              ...   
930  United States of America  Transportation support services   
931                     Japan                         Aluminum   
932     Taiwan, Greater China           Servers & data centers   
933                     Japan                 Engines & motors   
934  United States of America             Industrial machinery   

                                       Primary sector  \
0                      Commercial & consumer services   
1                        Medical equipment & supplies   
2                                  Financial services   
3                              Wood & paper materials   
4                                        Construction   
..                                                ...   
930                       Industrial support services   
931                Metal smelting, refining & forming   
932  Media, telecommunications & data center services   
933                                 Powered machinery   
934                                 Powered machinery   

                  Primary industry          business_travel  \
0                         Services                  2195.01   
1    Biotech, health care & pharma                    68909   
2                         Services                      518   
3                        Materials                      367   
4                   Infrastructure                     3973   
..                             ...                      ...   
930                       Services  Question not applicable   
931                      Materials  Question not applicable   
932                       Services  Question not applicable   
933                  Manufacturing  Question not applicable   
934                  Manufacturing                      304   

               capital_goods        downstream_assets  \
0    Question not applicable  Question not applicable   
1                    1878527  Question not applicable   
2    Question not applicable  Question not applicable   
3    Question not applicable  Question not applicable   
4                     255640  Question not applicable   
..                       ...                      ...   
930  Question not applicable  Question not applicable   
931  Question not applicable  Question not applicable   
932  Question not applicable  Question not applicable   
933  Question not applicable  Question not applicable   
934  Question not applicable  Question not applicable   

            downstream_trans  ...                 upstream  \
0    Question not applicable  ...  Question not applicable   
1    Question not applicable  ...  Question not applicable   
2    Question not applicable  ...  Question not applicable   
3                     215947  ..

## Status

##### As of right now, there is a data frame called 'df_rev_merge' it contains
935 companies that have correctly merged data, that is of the original 5913, so let me first go back and add in all the companies that I lost and then add the revenue data as "Missing" for the companies that I do have missing

In [118]:
for col in pivot_df.columns:
  print(col)

Account number
Organization
Country
Primary activity
Primary sector
Primary industry
business_travel
capital_goods
downstream_assets
downstream_trans
employee_comute
end_of_prod_life
franchises
non_scope_1_2_energy
investment
downstream
upstream
precessing_products
purchased_goods_and_services
upstream_assets
upstream_trans
sold_product_use
operations_waste


In [132]:
# Adding back the missing companies
df_all = pivot_df.join(df_rev_merge.set_index('Organization'), on='Organization', lsuffix='x', rsuffix='y')
df_all.count()

Account numberx                  5924
Organization                     5924
Countryx                         5924
Primary activityx                5922
Primary sectorx                  5922
Primary industryx                5922
business_travelx                 5895
capital_goodsx                   5897
downstream_assetsx               5905
downstream_transx                5895
employee_comutex                 5894
end_of_prod_lifex                5906
franchisesx                      5903
non_scope_1_2_energyx            5897
investmentx                      5902
downstreamx                      5913
upstreamx                        5910
precessing_productsx             5909
purchased_goods_and_servicesx    5892
upstream_assetsx                 5904
upstream_transx                  5897
sold_product_usex                5898
operations_wastex                5870
Account numbery                   935
Countryy                          935
Primary activityy                 935
Primary sect

In [133]:
# Rearranging the columns so that I can see if the names Match
cols = list(df_all.columns.values)
cols.pop(cols.index('Organization'))
cols.pop(cols.index('Company Name'))
df_all = df_all[['Organization', 'Company Name'] + cols]
df_all.head(10)

Organization                  Company Name  \
0                       3i Group                           NaN   
1       Abertis Infraestructuras                           NaN   
2             ABM INDUSTRIES INC          ABM Industries, Inc.   
3            Abbott Laboratories           Abbott Laboratories   
4                A T Kearney Ltd                           NaN   
5                 Aareal Bank AG                Aareal Bank AG   
6                          Acron                           NaN   
7  Resolute Forest Products Inc.  Resolute Forest Products Inc   
8             ACI WORLDWIDE  LTD                           NaN   
9        Aboitiz Equity Ventures                           NaN   

   Account numberx                                           Countryx  \
0               44  United Kingdom of Great Britain and Northern I...   
1               53                                              Spain   
2               58                           United States of America   
3               64                           United States of America   
4               79  United Kingdom of Great Britain and Northern I...   
5               97                                            Germany   
6              118                                 Russian Federation   
7              119                                             Canada   
8              123  United Kingdom of Great Britain and Northern I...   
9              134                                        Philippines   

                    Primary activityx  \
0                      Asset managers   
1  Infrastructure upkeep & management   
2                 Commercial services   
3                   Medical equipment   
4         Other professional services   
5                               Banks   
6             Nitrogenous fertilizers   
7                  Pulp & paper mills   
8                            Software   
9                Electricity networks   

                           Primary sectorx              Primary industryx  \
0                       Financial services                       Services   
1  Land & property ownership & development                 Infrastructure   
2           Commercial & consumer services                       Services   
3             Medical equipment & supplies  Biotech, health care & pharma   
4        Specialized professional services                       Services   
5                       Financial services                       Services   
6                                Chemicals                      Materials   
7                   Wood & paper materials                      Materials   
8                IT & software development                       Services   
9                  Energy utility networks                 Infrastructure   

          business_travelx           capital_goodsx       downstream_assetsx  \
0                     40.2  Question not applicable  Question not applicable   
1                 22581.41  Question not applicable  Question not applicable   
2                  2195.01  Question not applicable  Question not applicable   
3                    68909                  1878527  Question not applicable   
4                 21714.11                        0  Question not applicable   
5                      518  Question not applicable  Question not applicable   
6  Question not applicable  Question not applicable  Question not applicable   
7                      367  Question not applicable  Question not applicable   
8  Question not applicable  Question not applicable  Question not applicable   
9  Question not applicable  Question not applicable  Question not applicable   

   ...              downstreamy                upstreamy  \
0  ...                      NaN                      NaN   
1  ...                      NaN                      NaN   
2  ...  Question not applicable  Question not applicable   
3  ...  Question not applicable  Question not applicable   
4  ...  

In [134]:
# Rearranging the columns so that I can see if the IDS Match
cols = list(df_all.columns.values)
cols.pop(cols.index('Account numberx'))
cols.pop(cols.index('Account numbery'))
df_all = df_all[['Account numberx', 'Account numbery'] + cols]
df_all.head(10)

Account numberx  Account numbery                   Organization  \
0               44              NaN                       3i Group   
1               53              NaN       Abertis Infraestructuras   
2               58             58.0             ABM INDUSTRIES INC   
3               64             64.0            Abbott Laboratories   
4               79              NaN                A T Kearney Ltd   
5               97             97.0                 Aareal Bank AG   
6              118              NaN                          Acron   
7              119            119.0  Resolute Forest Products Inc.   
8              123              NaN             ACI WORLDWIDE  LTD   
9              134              NaN        Aboitiz Equity Ventures   

                   Company Name  \
0                           NaN   
1                           NaN   
2          ABM Industries, Inc.   
3           Abbott Laboratories   
4                           NaN   
5                Aareal Bank AG   
6                           NaN   
7  Resolute Forest Products Inc   
8                           NaN   
9                           NaN   

                                            Countryx  \
0  United Kingdom of Great Britain and Northern I...   
1                                              Spain   
2                           United States of America   
3                           United States of America   
4  United Kingdom of Great Britain and Northern I...   
5                                            Germany   
6                                 Russian Federation   
7                                             Canada   
8  United Kingdom of Great Britain and Northern I...   
9                                        Philippines   

                    Primary activityx  \
0                      Asset managers   
1  Infrastructure upkeep & management   
2                 Commercial services   
3                   Medical equipment   
4         Other professional services   
5                               Banks   
6             Nitrogenous fertilizers   
7                  Pulp & paper mills   
8                            Software   
9                Electricity networks   

                           Primary sectorx              Primary industryx  \
0                       Financial services                       Services   
1  Land & property ownership & development                 Infrastructure   
2           Commercial & consumer services                       Services   
3             Medical equipment & supplies  Biotech, health care & pharma   
4        Specialized professional services                       Services   
5                       Financial services                       Services   
6                                Chemicals                      Materials   
7                   Wood & paper materials                      Materials   
8                IT & software development                       Services   
9                  Energy utility networks                 Infrastructure   

          business_travelx           capital_goodsx  ...  \
0                     40.2  Question not applicable  ...   
1                 22581.41  Question not applicable  ...   
2                  2195.01  Question not applicable  ...   
3                    68909                  1878527  ...   
4                 21714.11                        0  ...   
5                      518  Question not applicable  ...   
6  Question not applicable  Question not applicable  ...   
7                      367  Question not applicable  ...   
8  Question not applicable  Question not applicable  ...   
9  Question not applicable  Question not applicable  ...   

               downstreamy                upstreamy     precessing_productsy  \
0                      NaN                      NaN                      NaN   
1                      NaN                      NaN                      NaN   
2  Question not applicable  

# Status
#### It looks like I know have a data frame with 935 revenue numbers all all the other data correct, I would now like to drop some of the things that I do not need to shrink it down to a smaller size and drop all the extra rows that I do not need

In [135]:
for col in df_all.columns:
  print(col)

Account numberx
Account numbery
Organization
Company Name
Countryx
Primary activityx
Primary sectorx
Primary industryx
business_travelx
capital_goodsx
downstream_assetsx
downstream_transx
employee_comutex
end_of_prod_lifex
franchisesx
non_scope_1_2_energyx
investmentx
downstreamx
upstreamx
precessing_productsx
purchased_goods_and_servicesx
upstream_assetsx
upstream_transx
sold_product_usex
operations_wastex
Countryy
Primary activityy
Primary sectory
Primary industryy
business_travely
capital_goodsy
downstream_assetsy
downstream_transy
employee_comutey
end_of_prod_lifey
franchisesy
non_scope_1_2_energyy
investmenty
downstreamy
upstreamy
precessing_productsy
purchased_goods_and_servicesy
upstream_assetsy
upstream_transy
sold_product_usey
operations_wastey
Data Year
Revenue (in $USD)


In [138]:
df_all.drop(['Account numbery','Countryy', 'Primary activityy', 'Primary sectory', 'Primary industryy', 'business_travely', 'capital_goodsy', 'downstream_assetsy', 'downstream_transy', 'employee_comutey', 'end_of_prod_lifey', 'franchisesy', 'non_scope_1_2_energyy', 'investmenty', 'downstreamy', 'upstreamy', 'precessing_productsy', 'purchased_goods_and_servicesy', 'upstream_assetsy', 'upstream_transy', 'sold_product_usey', 'operations_wastey', 'Data Year'],axis=1,inplace=True)


KeyError: ignored

In [139]:
df_all.head(5)

Account numberx              Organization          Company Name  \
0               44                  3i Group                   NaN   
1               53  Abertis Infraestructuras                   NaN   
2               58        ABM INDUSTRIES INC  ABM Industries, Inc.   
3               64       Abbott Laboratories   Abbott Laboratories   
4               79           A T Kearney Ltd                   NaN   

                                            Countryx  \
0  United Kingdom of Great Britain and Northern I...   
1                                              Spain   
2                           United States of America   
3                           United States of America   
4  United Kingdom of Great Britain and Northern I...   

                    Primary activityx  \
0                      Asset managers   
1  Infrastructure upkeep & management   
2                 Commercial services   
3                   Medical equipment   
4         Other professional services   

                           Primary sectorx              Primary industryx  \
0                       Financial services                       Services   
1  Land & property ownership & development                 Infrastructure   
2           Commercial & consumer services                       Services   
3             Medical equipment & supplies  Biotech, health care & pharma   
4        Specialized professional services                       Services   

  business_travelx           capital_goodsx       downstream_assetsx  ...  \
0             40.2  Question not applicable  Question not applicable  ...   
1         22581.41  Question not applicable  Question not applicable  ...   
2          2195.01  Question not applicable  Question not applicable  ...   
3            68909                  1878527  Question not applicable  ...   
4         21714.11                        0  Question not applicable  ...   

               investmentx              downstreamx                upstreamx  \
0  Question not applicable  Question not applicable  Question not applicable   
1  Question not applicable  Question not applicable  Question not applicable   
2  Question not applicable  Question not applicable  Question not applicable   
3  Question not applicable  Question not applicable  Question not applicable   
4  Question not applicable  Question not applicable  Question not applicable   

      precessing_productsx purchased_goods_and_servicesx  \
0  Question not applicable                        2388.5   
1  Question not applicable                     616060.76   
2  Question not applicable       Question not applicable   
3  Question not applicable                       6580577   
4  Question not applicable                       1691.96   

          upstream_assetsx          upstream_transx        sold_product_usex  \
0  Question not applicable  Question not applicable  Question not applicable   
1  Question not applicable  Question not applicable              16174433.43   
2  Question not applicable  Question not applicable  Question not applicable   
3  Question not applicable                  1610768                   508491   
4  Question not applicable  Question not applicable  Question not applicable   

         operations_wastex Revenue (in $USD)  
0                    16.67               NaN  
1                 26357.61               NaN  
2  Question not applicable      5.987600e+09  
3                    15088      3.460800e+10  
4                    82.31               NaN  

[5 rows x 25 columns]

# Status
#### I now have a smaller frame, that contains most of the information that I am going to need, I now need to try to match some more of the revenue data, this can now be done with either of the two companies names, there are 935 currently matched values, so there is a lot more matching that needs to be done.